core/htmlparser/filter.js

```diff
@@ -402,6 +402,164 @@
 } )();
 
 /**
+ * Abstract class describing the definition of {@link CKEDITOR.htmlParser.filter} rules.
+ *
+ * Definition object represents rules as a set of properties with callback functions
+ * to be applied for transforming and filtering content upon data processing.
+ *
  * @class CKEDITOR.htmlParser.filterRulesDefinition
  * @abstract
  */
+
+/**
+ * @property {CKEDITOR.htmlParser.nameFilterRule[]} elementNames An array of rules for element names transformation.
```

Comment:
TBH I do not like how this custom type looks like. I would think about replacing it with `Array.<RexExp,String>` syntax (note the dot that magically fixes jsduck issues; it would be also consistent with `Object.<whatever>` syntax that we use elsewhere).

Reply:
Yes it was my mistake, to skip the dot, yet it is an array of arrays, which is not correctly parsed by jsduck.
`Array.<RegExp,String>[]` -> `Array.&ltRegExp,String>[]`